templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,427 @@
+    fn parse_deploy_module_event(
+        &self,
+        block_item: BlockItemSummary,
+    ) -> Result<ModuleDeployed, DeployError> {
+        match block_item.details {
+            BlockItemSummaryDetails::AccountTransaction(a) => match a.effects {
+                AccountTransactionEffects::None {
+                    transaction_type,
+                    reject_reason,
+                } => {
+                    if transaction_type != Some(TransactionType::DeployModule) {
+                        return Err(DeployError::InvalidBlockItem(
+                            "Expected transaction type to be DeployModule".into(),
+                        ));
+                    }
+
+                    match reject_reason {
+                        RejectReason::ModuleHashAlreadyExists {
+                            contents,
+                        } => Ok(ModuleDeployed {
+                            module_ref: contents,
+                        }),
+                        _ => Err(DeployError::TransactionRejectedR(format!(
+                            "Module deploy rejected with reason: {reject_reason:?}"
+                        ))),
+                    }
+                }
+                AccountTransactionEffects::ModuleDeployed {
+                    module_ref,
+                } => Ok(ModuleDeployed {
+                    module_ref,
+                }),
+                _ => Err(DeployError::InvalidBlockItem("Invalid transaction effects".into())),
+            },
+            _ => Err(DeployError::InvalidBlockItem("Expected Account transaction".into())),
+        }
+    }
+
+    /// A function to parse the initialization events.
+    fn parse_contract_init_event(
+        &self,
+        block_item: BlockItemSummary,
+    ) -> Result<ContractInitialized, DeployError> {
+        match block_item.details {
+            BlockItemSummaryDetails::AccountTransaction(a) => match a.effects {
+                AccountTransactionEffects::None {
+                    transaction_type,
+                    reject_reason,
+                } => {
+                    if transaction_type != Some(TransactionType::InitContract) {
+                        return Err(DeployError::InvalidBlockItem(
+                            "Expected transaction type to be InitContract".into(),
+                        ));
+                    }
+
+                    Err(DeployError::TransactionRejectedR(format!(
+                        "Contract init rejected with reason: {reject_reason:?}"
+                    )))
+                }
+                AccountTransactionEffects::ContractInitialized {
+                    data,
+                } => Ok(ContractInitialized {
+                    contract: data.address,
+                }),
+                _ => Err(DeployError::InvalidBlockItem("Invalid transaction effects".into())),
+            },
+            _ => Err(DeployError::InvalidBlockItem("Expected Account transaction".into())),
```

Comment:
These errors do not help the user that much. Could we elaborate a bit on what might have gone wrong?

Reply:
Same below.

Reply:
I am using the following error text now:


```
The parsed account transaction effect should be of type `ContractInitialized` or `None` (in case the transaction reverted)
```
                     
and 
```
Can only parse an account transaction (no account creation transaction or chain update transaction)
```